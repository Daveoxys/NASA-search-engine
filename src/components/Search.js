import React, { useState } from "react";
import "../styles/Search.css";
import propTypes from "prop-types";
import getImages from "../requests/getImages";

const Search = ({ setSearchResults }) => {
  const [value, setValue] = useState();
  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(value));
  };

  return (
    <div className="Search">
      <form className="search-form" onSubmit={handleSubmit}>
        <input
          className="search-input"
          type="text"
          onChange={(e) => setValue(e.target.value)}
        />
        <button className="search-button" type="submit">
          Search
        </button>
      </form>
    </div>
  );
};

Search.propTypes = {
  setSearchResults: propTypes.func,
};

export default Search;
