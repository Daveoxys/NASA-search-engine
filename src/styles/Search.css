.search-input {
  margin: auto;
  width: 20%;
  height: 50px;
  border-radius: 15px;
  font-size: 20px;
  padding: 0 10px;
  border: 3px solid rgb(1, 1, 75);
}

.search-form {
  margin: auto;
}

.search-b {
  border-radius: 6px;
  border: 3px solid rgb(1, 1, 75);
}
