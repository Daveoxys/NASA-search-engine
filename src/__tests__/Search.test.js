import React, { render } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  const { asFragment } = render(<Search />);

  it("renders corrctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
