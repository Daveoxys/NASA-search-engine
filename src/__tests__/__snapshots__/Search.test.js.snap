// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders corrctly 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <form
      class="search-form"
    >
      <input
        class="search-input"
        type="text"
      />
       
      <br />
       
      <br />
      <button
        class="search-b"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
